base/applications/taskmgr/graphctl.h

```diff
@@ -22,93 +22,61 @@
 
 #pragma once
 
-#define MAX_PLOTS 4
-#define MAX_CTRLS 4
+#define NUM_PLOTS    2
+#define PLOT_SHIFT   2
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#if 0
-
-/* Attributes */
-public:
-  void SetXUnits(const char* string);
-  void SetYUnits(const char* string);
-
-  /* Operations */
-public:
-  BOOL Create(DWORD dwStyle, const RECT& rect, HWND hParentWnd, UINT nID=NULL);
+extern WNDPROC OldGraphCtrlWndProc;
 
-#endif
+INT_PTR CALLBACK GraphCtrl_WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam);
 
-typedef struct
+typedef struct _TM_GRAPH_CONTROL
 {
-  int m_nShiftPixels;          /* amount to shift with each new point */
-  int m_nYDecimals;
-
-  char m_strXUnitsString[50];
-  char m_strYUnitsString[50];
-
-  COLORREF m_crBackColor;                 /* background color */
-  COLORREF m_crGridColor;                 /* grid color */
-  COLORREF m_crPlotColor[MAX_PLOTS];      /* data color   */
-
-  double m_dCurrentPosition[MAX_PLOTS];   /* current position */
-  double m_dPreviousPosition[MAX_PLOTS];  /* previous position */
-
-/* those were protected fields */
-  int m_nHalfShiftPixels;
-  int m_nPlotShiftPixels;
-  int m_nGridOffsetPixels;
-  int m_nClientHeight;
-  int m_nClientWidth;
-  int m_nPlotHeight;
-  int m_nPlotWidth;
-  int m_nGridXPixels;
-  int m_nGridYPixels;
-
-  double m_dLowerLimit;        /* lower bounds */
-  double m_dUpperLimit;        /* upper bounds */
-  double m_dRange;
-  double m_dVerticalFactor;
-
-  HWND     m_hWnd;
-  HWND     m_hParentWnd;
-  HDC      m_dcGrid;
-  HDC      m_dcPlot;
-  HBITMAP  m_bitmapOldGrid;
-  HBITMAP  m_bitmapOldPlot;
-  HBITMAP  m_bitmapGrid;
-  HBITMAP  m_bitmapPlot;
-  HBRUSH   m_brushBack;
-  HPEN     m_penPlot[MAX_PLOTS];
-  RECT     m_rectClient;
-  RECT     m_rectPlot;
-} TGraphCtrl;
+    HWND     hParentWnd;
+    HWND     hWnd;
+    HDC      hdcGraph;
+    HBITMAP  hbmGraph;
+    HPEN     hPenGrid;
+    HPEN     hPen0;
+    HPEN     hPen1;
```

Comment:
If one for each plot, then maybe:
```suggestion
    HPEN     hPen[NUM_PLOTS];
```

Reply:
why? for the sake of changes? all taskmanager graphs have 2 plots at maximum - one, main, shown always, and second maybe, for some, for CPU namely. that array there is not needed and won't be.

Reply:
> for the sake of changes?

Says the one who has made quite a lot of changes beyond the original intent of the PR (fixing the graphs histories on resize).

> for CPU namely. that array there is not needed and won't be.

But the two pen members would still be there anyway, so...

Reply:
> Says the one who has made quite a lot of changes beyond the original intent of the PR (fixing the graphs > histories on resize).

I stated the main goal of the PR and didn't include bragging about removing BS, that resulted in more optimized work. originally, there were 4 plots and they were drawn allways, even if there was nothing to draw, look at `GraphCtrl_DrawPoint` for the example. indeed, there was so much cruft over there, that adding absolutely missing functionality (because the previous code was busy playing cool in mimicking OOP), it was still possible to remove twice as much of code as was added. I don't want to talk about the mess there and the code "style" btw as well.

> But the two pen members would still be there anyway, so...

so, dealing with them as separate members is easier and more comfortable, than wonking with unneeded array.